compiler/damlc/daml-preprocessor/src/DA/Daml/Preprocessor.hs

```diff
@@ -141,7 +141,7 @@ checkRecordConstructor (GHC.L _ m) = mapMaybe getRecordError (GHC.hsmodDecls m)
 
     message tyNameStr conNameStr = unwords
         [ "Record type", tyNameStr, "has constructor", conNameStr
-        , "with different name. This may cause problems with cross-SDK upgrades."
+        , "with different name."
         , "Possible solution: Change the constructor name to", tyNameStr ]
```

Comment:
Isn't that pretty much the only solution now?

Reply:
No you can also change the type name, turn it into an enum in some cases, …